### ISSUE TITLE
Add Q&A Group to FE template of Resources & Support Detail page

### ISSUE THREAD

**Issue by wesrowe** [department-of-veterans-affairs/va.gov-cms#9755]
## Description

As a Veteran, I want to see Q&A Groups composed of reusable Q&As on the R&S Detail Page type, so that the content I see is more easily governed under a single source of truth (reusable) model.

### Details

The CMS was enabled to add Q&A groups (with reusable Q&As) to the R&S Detail page in department-of-veterans-affairs/va.gov-team#8643. In this issue, we will enable the front-end template to display that data.

See department-of-veterans-affairs/va.gov-team#9579 for discovery and technical details.

### Design

Generally we are following the same layout as Q&A Groups that appear in other templates, e.g., on FAQ pages. 

Observations of existing FAQ pages:
- The "section header" field of Q&A Groups is frequently left blank - [example](https://www.va.gov/resources/montgomery-gi-bill-refunds/)
- Two header-less Q&A Groups can be placed on the same page (not sure why)
  - "Signing into VA.gov" page [on Prod](https://www.va.gov/resources/signing-in-to-vagov/)
  - That page in [CMS](https://prod.cms.va.gov/node/8222/edit)

Data:
- Q&A Group header/title
- Rich-text description paragraph
- Individual Q&A entities

### Tasks
Assumptions: 
- Drupal eng will deliver data fields to be displayed
- presentation will mirror Q&A Groups on FAQ templates. No new styles, just maybe some variations fields being handled.
- Fields will include: header (optional), description (optional), and potential for multiple Questions/Answers.
- PR merge will be blocked by department-of-veterans-affairs/va.gov-team#9737 but implementation isn't (can work in parallel)
- [ ] Review FAQ implementation of Q&As - 2-3hrs
- [ ] Port data to R&S template - 1-3hrs
- [ ] Check for existing unit tests - .5hr
- [ ] Create unit tests - should prob be separate ticket
- [ ] update unit tests - ~.5hr
- [ ] Test locally - 1-2hr
- [ ] Design review with Florence - 1hr
- [ ] Potentially: changes from design review - ?TBD? 1h-1/2day
- [ ] PR draft / code review & changes / shepherd to merge - .5/1h

7.5h - 15h (1.5-3days)

## Acceptance Criteria
- [ ] All Q&A data entered by Editor is rendered on the FE in accordance with Design System
- [ ] Layout handles case when section header is missing
- [ ] Layout handles case when paragraph is missing
- [ ] Requires design review

### CMS Team
Please check the team(s) that will do this work.

- [ ] `Program`
- [ ] `Platform CMS Team`
- [ ] `Sitewide Crew`
- [ ] `⭐️ Sitewide CMS`
- [x] `⭐️ Public Websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

**Comment by wesrowe** [comment]
@jcklpe (cc @jilladams), based on the assumption that Q&A Groups on R&S Details pages should look just like FAQ pages, I just started diving into the existing FAQ pages. I immediately found some odd use cases that I think need follow-up with the Editors. 

1. Why is it hard to find an FAQ where the Section Header of the Q&A Group is actually filled in? (hypothesis: there isn't much content between the `h1` and the start of Q&A Groups, usually just a paragraph and 1 or 2 CTA links.) - [Example](https://www.va.gov/resources/montgomery-gi-bill-refunds/)
2. Why in [this example](https://prod.cms.va.gov/node/8222/edit) would you have Q&As broken into two Q&A Groups with no distinction visible to the user? (i.e., both section headers were left blank)

**Comment by jilladams** [comment]
Noted in mid-sprint email to Dave that this ticket is stretch and may not complete in Sprint 63, depending on when GraphQL changes are available to work with.

**Comment by jilladams** [comment]
@laflannery noted that Q&A have a 508 heading issue, where Question title is same heading level as the accordion itself. That may already be ticketed elsewhere, TBD, Laura is digging. If we surface it in time and fix is small enough to include a fix in this PR: yay. If not, we'll plan for it as a fast follow.

**Comment by laflannery** [comment]
@jilladams The issue I was referring to is in the team repo [here. ](https://github.com/department-of-veterans-affairs/va.gov-team/issues/30359)I think it could use some more description (i.e. I know of 1 instance where the headings are the same and I know of another instance where the section heading is h3 and accordions are h2).  Would you like me to assign this ticket to you, create a new one, do nothing for now since it may be slightly more involved?

**Comment by jtmst** [comment]
this is blocked until the errors on https://github.com/department-of-veterans-affairs/content-build/pull/1253 are figured out and resolved.

**Comment by chri5tia** [comment]
FYI I closed #9901 with a note. In retrospect, that ticket should have been closed once I pasted in my version of the GraphQl query to illustrate the content model changes. And now that I know about the air table spec tool, we probably didn't need to go about it that way at all. In the future, I'll just do the GraphQL explorer test as part of my regular CMS work when it touches FE. Let me know if you have any questions.

**Comment by wesrowe** [comment]
To clarify the headings issue – We've determined fixing accordion heading levels is outside the scope of this issue. Another issue has been created to discover how this will be done, hopefully using the design system: #10136.

@chri5tia, can you please tick the AC checkboxes as appropriate?

**Comment by chri5tia** [comment]
Working on the "Review FAQ Implementation of Q&As" peice in tandem with Josh. We have run into a discrepancy and trying to unravel how the reusable Q&A Group was/if it was ever working on FAQ pages in order to find out why the query to pull in the answers if failing.

Dumping some likely intelligible notes here for now...

Working backwards through an example...

Single Q&A Node:
https://va-gov-cms.ddev.site/resources/how-do-i-sign-in-to-vagov

NID = 8208
Content type = Q&A - single

A page that uses this single reusable Q&A node:
https://va-gov-cms.ddev.site/resources/signing-in-to-vagov NID = 8222
Content type = FAQ

Key differences between FAQ and R&S:
- Q&A Group is added via "add content" block which adds the entity reference
- FAQ page has a field as part of the content type for the entity reference

![Screen Shot 2022-08-18 at 11 52 16 AM](https://user-images.githubusercontent.com/22764938/185472021-828c576b-a5a6-4d34-b260-1aa33242edb6.png)

The way that Q&A Group is made available for this entity reference is the same: entity reference field settings. 

Draft query to expose the Q&As on R&S pages for specific node doesn't expose the answers piece of the single Q&A node.

**Comment by jilladams** [comment]
PR https://github.com/department-of-veterans-affairs/content-build/pull/1256
